src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -29,9 +29,9 @@ internal sealed class CSharpChangeSignatureService : AbstractChangeSignatureServ
             var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
             var token = tree.GetRoot(cancellationToken).FindToken(position != tree.Length ? position : Math.Max(0, position - 1));
 
-            var ancestorDeclarationKinds = restrictToDeclarations ? _invokableAncestorKinds.Add(SyntaxKind.Block) : _invokableAncestorKinds;
+            var ancestorDeclarationKinds = restrictToDeclarations ? _invokableAncestorKinds.Add(SyntaxKind.Block).Add(SyntaxKind.ArrowExpressionClause) : _invokableAncestorKinds;
             SyntaxNode matchingNode = token.Parent.AncestorsAndSelf().FirstOrDefault(n => ancestorDeclarationKinds.Contains(n.Kind()));
-            if (matchingNode == null || matchingNode.IsKind(SyntaxKind.Block))
+            if (matchingNode == null || matchingNode.IsKind(SyntaxKind.Block) || matchingNode.IsKind(SyntaxKind.ArrowExpressionClause))
```

Comment:
`Microsoft.CodeAnalysis.CSharp.Extensions` has an overload to `IsKind` that takes many parameters. So this could also be `.IsKind(Block, Arrow)`.